Implemented Financial Report Generator

minghan36 opened the pull request:
# Implemented Financial Report Generator

## Related Issue
Closes #78 

## Description

- New 'Generate Summary' button on dashboard view.
- Clicking on button generates a pdf file at path `src\main\resources\static\summary.pdf`.
- Summary automatically opens in a new tab.

## Testing

- [x] Does the 'Generate Summary' button exist in the Dashboard view?
- [x] Is the summary pdf generated and opened in a new tab?
- [x] Are values across the summary consistent? (Total Expenses, Total Income, Net Cashflow)
- [x] Do the values change as a new transaction is added?

## Testing
![Capture](https://github.com/user-attachments/assets/067e35a6-22b6-446d-b999-bf473f9c1c19)


## Additional Context
Use Username: student  Password: student to avoid needing to add many expenses and incomes.

minghan36 committed:
Feature: Financial Report Generator
Generates a pdf report for the current user. The pdf is displayed in a new tab once generated. Contains all transactions for a user.

minghan36 committed:
Feature: Generate PDF Summary
Addition of new models to assist in generation of summary

minghan36 committed:
Chore: gitignore generated summary pdf

ck0x reviewed (changes requested):
### Testing
- All given tests are passed.

### Actions
- At the moment when adding an income at a current or future date, it is not taken into account in the generated report. From my understanding, transactions within the month should be taken into account.

Everything looks great, thanks for your work!

minghan36 committed:
Fix: Summary does not account of incomes in the present month
Changed to account for all incomes and expenses from the current month. Any transactions from future months are left out.
